driver/src/main/java/eu/cloudnetservice/driver/network/rpc/defaults/rpc/DefaultRPC.java

```diff
@@ -123,56 +146,87 @@ public DefaultRPC(
    */
   @Override
   public @NonNull Type expectedResultType() {
-    return this.expectedResultType;
+    return this.targetMethod.unwrappedReturnType();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public @NonNull RPCMethodMetadata targetMethod() {
+    return this.targetMethod;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public @NonNull RPC timeout(@Nullable Duration timeout) {
+    this.executionTimeout = timeout;
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public @Nullable Duration timeout() {
+    return this.executionTimeout;
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
-  public @NonNull RPC disableResultExpectation() {
-    this.resultExpectation = false;
+  public @NonNull RPC dropResult() {
+    this.dropResult = true;
     return this;
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
-  public boolean expectsResult() {
-    return this.resultExpectation;
+  public boolean resultDropped() {
+    return this.dropResult;
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void fireAndForget() {
-    this.fireAndForget(Objects.requireNonNull(this.sender.associatedComponent().firstChannel()));
+    var targetNetworkChannel = this.channelSupplier.get();
```

Comment:
Is it possible to share this with the chained implementation? (Also for the other fire methods)

Reply:
I mean, technically yes, but it just introduces a class for sharing a single exception case so I don't really see the advantage (aside from: if the error message is ever subject to change, you don't accidentally miss one place)